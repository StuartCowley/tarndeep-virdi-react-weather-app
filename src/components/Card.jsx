import Forecast from "./Forecast";
import React from "react";
import data from "../data/forecast.json";

export default function Card() {
  let cards;

  return <section className="cards">{cards}</section>;
}
