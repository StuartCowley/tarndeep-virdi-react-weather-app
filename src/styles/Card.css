.card--forecast {
  width: 20vw;
  max-width: 300px;
  text-align: center;
}

.cards * {
  border: 1px red solid;
}

.cards {
  display: flex;
  justify-content: center;
}

@media (max-width: 550px) {
  .card--forecast {
    display: flex;
    width: 100%;
    max-width: 100%;
    justify-content: space-between;
    max-height: 80px;
  }
  .cards {
    margin: 0 auto;
    width: 90vw;
    display: block;
  }
}
