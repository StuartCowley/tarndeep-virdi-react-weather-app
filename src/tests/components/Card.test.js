import React from "react";
import { render } from "@testing-library/react";
import Card from "../../components/Card";

describe("Card", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <Card
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  //   it("renders correct values for props", () => {
  //     const { getByText } = render(
  //       <Card
  //         date={validProps.date}
  //         description={validProps.description}
  //         icon={validProps.icon}
  //         temperature={validProps.temperature}
  //       />
  //     );
  //   expect(getByText("Thu")).toHaveAttribute("class", "card--day");
  //   expect(getByText("29 Aug")).toHaveAttribute("class", "card--date");
  //   expect(getByText("Stub description")).toHaveAttribute("class", "card--description");
  //   expect(getByText("22")).toHaveAttribute("class", "card--temp-max");
  //   expect(getByText("14")).toHaveAttribute("class", "card--temp-min");
  //   });
});
