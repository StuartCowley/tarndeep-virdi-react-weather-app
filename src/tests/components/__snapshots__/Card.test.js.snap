// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<DocumentFragment>
  <section
    class="cards"
  >
    <div
      class="card--forecast"
    >
      <div
        class="card--daydate"
      >
        <p>
          Monday
        </p>
        <p>
          18/10
        </p>
      </div>
      <p>
        Clear
      </p>
      <div
        class="card--temp"
      >
        <h4>
          11
        </h4>
        <h5>
          4
        </h5>
      </div>
      <p>
        icon
      </p>
    </div>
    <div
      class="card--forecast"
    >
      <div
        class="card--daydate"
      >
        <p>
          Tuesday
        </p>
        <p>
          15/07
        </p>
      </div>
      <p>
        Stormy
      </p>
      <div
        class="card--temp"
      >
        <h4>
          13
        </h4>
        <h5>
          8
        </h5>
      </div>
      <p>
        icon
      </p>
    </div>
    <div
      class="card--forecast"
    >
      <div
        class="card--daydate"
      >
        <p>
          Wednesday
        </p>
        <p>
          11/04
        </p>
      </div>
      <p>
        Heavy Snow
      </p>
      <div
        class="card--temp"
      >
        <h4>
          1
        </h4>
        <h5>
          -2
        </h5>
      </div>
      <p>
        icon
      </p>
    </div>
    <div
      class="card--forecast"
    >
      <div
        class="card--daydate"
      >
        <p>
          Thursday
        </p>
        <p>
          05/01
        </p>
      </div>
      <p>
        Tornado
      </p>
      <div
        class="card--temp"
      >
        <h4>
          20
        </h4>
        <h5>
          4
        </h5>
      </div>
      <p>
        icon
      </p>
    </div>
    <div
      class="card--forecast"
    >
      <div
        class="card--daydate"
      >
        <p>
          Friday
        </p>
        <p>
          02/10
        </p>
      </div>
      <p>
        Hazy
      </p>
      <div
        class="card--temp"
      >
        <h4>
          25
        </h4>
        <h5>
          18
        </h5>
      </div>
      <p>
        icon
      </p>
    </div>
  </section>
</DocumentFragment>
`;
